### ISSUE TITLE
`bun run` should prioritize running script over file/directory

### ISSUE THREAD

**Issue by PatrickMatthiesen** [oven-sh/bun#7735]
### What version of Bun is running?

1.0.18+1a2643520

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Simply have a script called `build` and try to run it with the `--bun` flag

### What is the expected behavior?

The script should run like the following:
![image](https://github.com/oven-sh/bun/assets/43612965/ee5e4ea6-6530-4e5d-bdc5-7da241e5c16f)
Note I removed the `d` in the script name to make it work

### What do you see instead?

![image](https://github.com/oven-sh/bun/assets/43612965/2e76795c-3220-4ffa-bdd0-c78723a3dfbf)

### Additional information

I couldn't reproduce the issue with other flags but the --bun and -b flags.

**Comment by Electroid** [comment]
Interesting, tried on macOS arm64.

```
> bun run --bun build
$ echo Hi
Hi
```

```json
{
  "dependencies": {
    "express": "^4.18.2"
  },
  "scripts": {
    "build": "echo Hi"
  }
}
```

**Comment by PatrickMatthiesen** [comment]
I did some more testing and figured out that it's because I have a directory that is named build. 
So it is an issue if a script and folder in the path have the same name. 
I would assume it's because Bun will prioritize the folder over the script name?

**Comment by Electroid** [comment]
Yep, looks like that's the issue. We should fix this so `bun run [name]` prioritizes running the script, before checking for a file/directory.

**Comment by paperclover** [comment]
ah, is it checking for an item named [name] and then finding the directory, then giving up?

actually no, we dont run the module resolver for script names

<img width="670" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/db375b1c-d18b-4499-ab63-3ebe08f1e988">

reminds me that i actually made this work if you run node as bun

<img width="375" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/6de1f560-9b9e-44b5-bf1f-2737575e5bad">

I dont know if the logic can be copy pasted from this but it likely is possible.

**Comment by paperclover** [comment]
idk if this is a goodfirstissue but if someone wants to investigate the code, run command starts in run_command.zig line 1032 "pub fn exec("

**Comment by aboqasem** [comment]
Debug logs:

<details>

![Debug logs](https://github.com/oven-sh/bun/assets/62098043/174a1370-6e98-4038-9f14-93a7129fc8ae)

</details>

**Comment by ryoppippi** [comment]
Is there any way to force bun to use script?

**Comment by RiskyMH** [comment]
Should be fixed by @pfgithub in #15117
